fgpyo/fasta/reference_set_builder.py

```diff
@@ -0,0 +1,174 @@
+"""
+Classes for generating Fasta files and records for testing
+----------------------------------------------------------------
+"""
+# import hashlib
+import os
+import textwrap
+from tempfile import NamedTemporaryFile
+from typing import List
+from typing import Optional
+
+# pylint: disable=W0511
+
+
+class ReferenceSetBuilder:
+    """
+    Builder for constructing one or more fasta records.
+    """
+
+    # The default asssembly
+    DEFAULT_ASSEMBLY: str = "testassembly"
+
+    # The default species
+    DEFAULT_SPECIES: str = "testspecies"
+
+    # Way to store instance of ReferenceBuilder
+    # TODO make something better than a list... probably
+    REF_BUILDERS: List["ReferenceBuilder"] = []
+
+    def __init__(
+        self,
+        assembly: Optional[str] = None,
+        species: Optional[str] = None,
+        line_length: int = 80,
+    ):
+        self.assembly: str = assembly if assembly is not None else self.DEFAULT_ASSEMBLY
+        self.species: str = species if species is not None else self.DEFAULT_SPECIES
+        self.line_length = line_length
+
+    def add(
+        self,
+        name: str,
+    ) -> "ReferenceBuilder":
+        """
+        Returns instance of ReferenceBuilder
+        """
+
+        builder = ReferenceBuilder(name=name, assembly=self.assembly, species=self.species)
+        self.REF_BUILDERS.append(builder)
+        return self.REF_BUILDERS[-1]
+
+    def writer_helper(
+        self,
+    ) -> None:
+        """ Place holder for helper funciton """
+        return None
+
+    def to_temp_file(
+        self,
+        delete_on_exit: Optional[bool] = True,
+        calculate_md5_sum: Optional[bool] = False,
+    ) -> None:
+        """
+        For each instance of ReferenceBuilder in REF_BUILDERS write record to temp file
+        """
+        # Write temp file path
+        path = NamedTemporaryFile(
+            prefix=f"{self.assembly}_{self.species}",
+            suffix=".fasta",
+            delete=delete_on_exit,
+            mode=("a+t"),
+        )
+
+        for record in range(len(self.REF_BUILDERS)):
+            seq = self.REF_BUILDERS[record].sequences
+            assembly = self.REF_BUILDERS[record].assembly
+            species = self.REF_BUILDERS[record].species
+            name = self.REF_BUILDERS[record].name
```

Comment:
`record` to me implies an instance of the builder, but here it's an index into the builders.  Either change it to `index` (then create a temporary variable named `builder = self.REF_BUILDERS[index]` so you don't have to type out the right side every time), or my preference, change it to `builder`
```suggestion
        for builder in self.REF_BUILDERS:
            sequences = builder.sequences
            assembly = builder.assembly
            species = builder.species
            name = builder.name
```